### DISCUSSION TITLE
How do I call the tool to return html or markdown, which can be displayed on the interface, and which contains a Button or a link that the user can click to interact with

### DISCUSSION THREAD

**Discussion by bigRabbit01** [open-webui/open-webui#6727] [General]
How do I call the tool to return html or markdown, which can be displayed on the interface, and which contains a Button or a link that the user can click to interact with

**Comment by bigRabbit01** [comment]
I have already added the Artifacts V2 function, but the interface does not render my html when I return it directly in the tool.

**Reply by peuportier** [reply]
if you have the last version, normally when the model make some html / css / java you shall have automatically the render otherwise just click on right top in the ... 
<img width="1221" alt="Screenshot 2024-11-05 at 17 05 00" src="https://github.com/user-attachments/assets/eee93d0e-07af-46cf-83e5-700ab9bb6711"> then select Artifacts

**Reply by bigRabbit01** [reply]
Thank you, I tried to call the tool to return an html, but the final return to the front-end interface is not the original html I generated, but the LLM converted into plain text

**Reply by bigRabbit01** [reply]
Can the calling tool return directly to the front end display without going through the LLM?

**Reply by GrayXu** [reply]
> Thank you, I tried to call the tool to return an html, but the final return to the front-end interface is not the original html I generated, but the LLM converted into plain text

It seems the HTML output, though supported in Markdown, isn't rendering correctly in Open WebUI and is instead displayed as plain text. This is an Open WebUI issue, not an LLM problem.
